backend/inspirehep/records/tasks.py

```diff
@@ -236,3 +242,40 @@ def reference_self_curation(
         record_revision_id=record.revision_id,
         user_email=user_email,
     )
+
+
+@shared_task(
+    retry_backoff=True,
+    acks_late=True,
+    retry_kwargs={"max_retries": 3},
+    autoretry_for=(*DB_TASK_EXCEPTIONS, *ES_TASK_EXCEPTIONS),
+)
+def export_legacy_recids(bucket_name, recids, db_batch_size):
+    current_s3_instance.create_bucket(bucket_name)
+    mime_type = "application/xml"
+    acl = current_app.config["S3_FILE_ACL"]
+    recid_len = len(recids)
+    upload_counter = 0
+    query = (
+        LegacyRecordsMirror.query.filter(LegacyRecordsMirror.recid.in_(recids))
+        .yield_per(db_batch_size)
+        .with_entities(LegacyRecordsMirror.recid, LegacyRecordsMirror._marcxml)
+    )
+    for legacy_record in query:
+        filename = str(legacy_record[0])
+        key = f"legacy_xml_{legacy_record[0]}"
+        fileob = BytesIO(legacy_record[1])
+        try:
+            current_s3_instance.upload_file(
+                fileob, key, filename, mime_type, acl, bucket_name
+            )
+            upload_counter += 1
+        except ClientError as e:
+            LOGGER.warning(
+                "There was an issue uploading to s3",
+                filename=filename,
+                bucket_name=bucket_name,
+                stacktrace=e,
+            )
+        if upload_counter % math.ceil(recid_len / 10) == 0:
+            LOGGER.info("files uploaded", upload_count=upload_counter)
```

Comment:
can't you just log in in the end of your CLI?

Reply:
Doesn't only the tasks know if the upload was successful? thats why i log it there.

Reply:
yes, but I meant to remove the check, why do you need that? If you process all the documents they're uploaded or am I wrong?

Reply:
No you are right, its just to see the upload progress of each task.

Reply:
I still don't understand why you can't just log `files are updated` at the end of the loop, without the `if` condition. Then, you don't need a counter (also, you can use enumerate, it's more "elegant" I would say), especially, since you will upload 100 files only in one batch.

Reply:
I will remove it. And I agree, enumerate would be more elegant but I think I would not be able to use it here because I only want to count the successful uploads.